divi/src/main.cpp

```diff
@@ -3292,13 +3292,13 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         CInv inv(MSG_TX, tx.GetHash());
         pfrom->AddInventoryKnown(inv);
 
-        LOCK(cs_main);
-
         bool fMissingInputs = false;
         CValidationState state;
 
         CNode::ClearInventoryItem(inv);
 
+        {
+        LOCK(cs_main);
```

Comment:
The main lock should probably cover the clearing of the global inventory items that we're keeping track of, just in case.

Reply:
Other than that, everything else looks good 👍

Reply:
Good point, updated.